src/chains/ethereum/console.log/scripts/helpers.ts

```diff
@@ -0,0 +1,337 @@
+ *
+ * import "console.sol";
+ *
+ * contract ConsoleLogTest {
+ *   function testLog(string memory value) public view {
+ *     console.log(value);
+ *   }
+ * }
+ * ```
+ *
+ * @returns
+ */
+export const createContract = (functions: FunctionDescriptor[]) => {
+  const functionStrings = functions.map(
+    ({ functionName, params, consoleSolLogFunctionToCall }) => {
+      const sigParams = params.map(({ type }, i) => `${type} a${i}`).join(", ");
+      const callParams = params.map((_, i) => `a${i}`).join(", ");
+      return `  function ${functionName}(${sigParams}) public view {
+    console.${consoleSolLogFunctionToCall}(${callParams});
+  }`;
+    }
+  );
+  return `// SPDX-License-Identifier: MIT
+pragma solidity >= 0.4.22 <0.9.0;
+
+import "console.sol";
+
+contract ${CONTRACT_NAME} {
+  function adversarialTest() public view {
+    bytes memory badHandler = abi.encodeWithSignature("fake()");
+    bytes memory badSignature = abi.encodeWithSignature("log(string)", -123456789);
+    address consolePrecompile = address(0x000000000000000000636F6e736F6c652e6c6f67);
+    address randomAddress = address(0x00000000000072616E646f6d2061646472657373);
+    assembly {
+      let badHandlerLength := mload(badHandler)
+      let badHandlerOffset := add(badHandler, 32)
+      pop(staticcall(gas(), consolePrecompile, badHandlerOffset, badHandlerLength, 0, 0)) // tests bad handler
+
+      let badSigLength := mload(badSignature)
+      let badSigOffset := add(badSignature, 32)
+      pop(staticcall(gas(), consolePrecompile, badSigOffset, badSigLength, 0, 0)) // tests bad data
+
+      pop(staticcall(gas(), consolePrecompile, 0, 0, 0, 0)) // tests short inLength
+      pop(staticcall(gas(), consolePrecompile, 0, 4, 0, 0)) // tests no valid data in memory
+      pop(staticcall(gas(), consolePrecompile, 999, 0, 0, 0)) // out of bounds memory read (offset)
+      pop(staticcall(gas(), consolePrecompile, 0, 999, 0, 0)) // out of bounds memory read (length)
+      pop(staticcall(gas(), randomAddress, 0, 0, 0, 0)) // ensure we don't listen to other contract addresses
+    }
+  }
+${functionStrings.join("\n\n")}
+}`;
+};
+
+export const compileContract = (contractSource: string) => {
+  const sources = {
+    "console.sol": {
+      content: consoleSol
+    }
+  };
+
+  const { contracts, errors } = JSON.parse(
+    solc.compile(
+      JSON.stringify({
+        language: "Solidity",
+        sources: {
+          [CONTRACT_NAME]: {
+            content: contractSource
+          },
+          ...sources
+        },
+        settings: {
+          outputSelection: {
+            "*": {
+              "*": ["*"]
+            }
+          }
+        }
+      })
+    )
+  );
+
+  /* istanbul ignore if */
+  if (errors && errors.some(error => error.severity === "error")) {
+    throw new Error(errors.map(e => e.formattedMessage).join("\n\n"));
+  }
+
+  const contract = contracts[CONTRACT_NAME][CONTRACT_NAME];
+  delete contracts[CONTRACT_NAME];
```

Comment:
Why delete [CONTRACT_NAME] here? are we not just discarding the `contracts` object

Reply:
for fun of course! (i've committed a fix for this)